frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -208,13 +209,74 @@ private void refreshModel(GitRepository gitRepository, List<String> skippedBranc
   }
 
   @UIEffect
-  private void openDiscoverDialog() {
-    var action = ActionManager.getInstance().getAction(ACTION_DISCOVER);
-    var dataContext = DataManager.getInstance().getDataContext(EnhancedGraphTable.this);
-    var anActionEvent = AnActionEvent.createFromDataContext(ACTION_PLACE_EMPTY_TABLE, new Presentation(), dataContext);
-    action.actionPerformed(anActionEvent);
+  private void doAutomaticDiscover(Path macheteFilePath) {
+    var selectedRepository = getGitRepositorySelectionProvider().getSelectedGitRepository().getOrNull();
+    if (selectedRepository == null) {
+      LOG.error("Can't do automatic discover because of undefined selected repository");
+      return;
+    }
+    var mainDirPath = GitVfsUtils.getMainDirectoryPath(selectedRepository).toAbsolutePath();
+    var gitDirPath = GitVfsUtils.getGitDirectoryPath(selectedRepository).toAbsolutePath();
+
+    @SuppressWarnings({"nullness", "guieffect"}) // To allow runResult be uninitialized and Tuple accept @Nullable IBranchLayout
```

Comment:
Hmm what's the specific error caused by `runResult` being uninitialized? :thinking: Is it about dereference of `runResult` in line 245?

Reply:
If so, then consider checking against `runResult == null` in `onSuccess`

Reply:
No, this is about this field is uninitialized by constructor: `error: [[fbc, nullness, allcheckers]:initialization.fields.uninitialized] the constructor does not initialize fields: runResult`

Reply:
Ahh right... then maybe mark this field as `@MonotonicNonNull`?

Reply:
Get rid of this field completely